### PR TITLE
fix: [PIE-6417]: removed useEffect to handle the repo change

### PR DESCRIPTION
**Pull request by sumanmabd**
### Summary

Removed the useEffect and added the branch with state update in the onChange repo handler for the filter
#### Screenshots
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx [hunk 1]
```diff
@@ -149,11 +149,9 @@ function ExecutionListInternal(props: ExecutionListProps): React.ReactElement {
   const showSpinner = initLoading || (loading && !isPolling)
 
   const onChangeRepo = (_repoName: string): void => {
+    setSelectedBranch(undefined)
```
Comment (douglasnfrazao) [inline src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx:152]: We don't need to verify anymore if repoName is null?

Comment (sumanmabd) [reply src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx:152]: lets say user selected a repo and removed the repo from the filter. now the repoName is null, and branch selected is also null. Fine right ? what ever the repo Name maybe , branch should be always should be set to null after the repo is changed in filter even if it is null.

**Review by douglasnfrazao — APPROVED** [review]
looks good

**Comment by sumanmabd** [comment]
retrigger coverage

**Review by udayvunnam — DISMISSED** [review]

**Comment by sumanmabd** [comment]
retrigger coverage

**Comment by sumanmabd** [comment]
retrigger cypress-cd

**Comment by sumanmabd** [comment]
retrigger coverage

**Review by chetanvekariya-harness — DISMISSED** [review]

**Comment by udayvunnam** [comment]
add coverage for the newly added function

**Review by sunnykesh-harness — CHANGES_REQUESTED** [review]

**Review by sunnykesh-harness — COMMENTED** [review]
Unblocking as this is solving few other important issue but please track the new issues created with this change. It will be good if all can be fixed together and we avoid any regression.

**Review by udayvunnam — DISMISSED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**